### PR TITLE
Make merge gatekeeper more resilient

### PR DESCRIPTION
**Pull request by franklad**
https://app.shortcut.com/clear-street/story/56657/make-merge-gatekeeper-more-resilient

### CODE REVIEW

File: internal/github/retry_roundtripper.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+	for i := 0; i <= t.Retries; i++ {
+		resp, err = t.Transport.RoundTrip(req)
+		if err != nil {
+			fmt.Println("Error:", err)
+			continue
+		}
+
+		if resp.StatusCode >= 500 && resp.StatusCode < 600 {
+			fmt.Printf("Retrying request due to status code: %d\n", resp.StatusCode)
```
Comment (jdechicchis) [inline internal/github/retry_roundtripper.go]: Does merge gatekeeper use some logging framework already? If so, I think we should use it

File: internal/github/retry_roundtripper.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+func (t *RetryTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	var (
+		resp *http.Response
+		err  error
+	)
+
+	for i := 0; i <= t.Retries; i++ {
+		resp, err = t.Transport.RoundTrip(req)
+		if err != nil {
+			fmt.Println("Error:", err)
```
Comment (jdechicchis) [inline internal/github/retry_roundtripper.go]: Make this log similar to the one below. For example: `"Retrying request due to error: %v", err`

File: internal/cli/validate.go [hunk 3]
```diff
@@ -72,6 +79,8 @@ func validateCmd() *cobra.Command {
 
 	cmd.PersistentFlags().StringVarP(&ignoredJobs, "ignored", "i", "", "set ignored jobs (comma-separated list)")
 
+	cmd.PersistentFlags().IntVar(&githubClientRetry, "client-retry", 0, "set retry count for github client")
```
Comment (jdechicchis) [inline internal/cli/validate.go]:
```suggestion
	cmd.PersistentFlags().IntVar(&githubClientRetry, "github-client-retry", 0, "set retry count for GitHub client")
```

File: internal/cli/validate.go [hunk 4]
```diff
@@ -72,6 +79,8 @@ func validateCmd() *cobra.Command {
...
+
```
Comment (jdechicchis) [inline internal/cli/validate.go:83]: Update docs with this option: https://github.com/clear-street/merge-gatekeeper#-action-inputs

Comment (jdechicchis) [inline internal/cli/validate.go:83]: Need to update the action yaml with this input option as well https://github.com/clear-street/merge-gatekeeper/blob/main/action.yml

**Review by jdechicchis — APPROVED** [review]

**Review by kanishkydv — APPROVED** [review]

**Pull request merged**